src/test/scala/de/zalando/beard/renderer/BeardTemplateRendererSpec.scala

```diff
@@ -54,7 +54,18 @@ class BeardTemplateRendererSpec extends FunSpec with Matchers {
         val renderResult = StringWriterRenderResult()
 
         renderer.render(template, renderResult, Map("users" -> Seq(Map("name" -> "Gigi"))))
-        renderResult.result.toString should be("<div>Hello</div>")
+        renderResult.result.toString should be("<div>HelloGigi</div>")
+      }
+
+      it("should render a template with a for statement with an index") {
+        val template = BeardTemplateParser {
+          Source.fromInputStream(getClass.getResourceAsStream(s"/templates/for-index-statement.beard")).mkString
+        }
+
+        val renderResult = StringWriterRenderResult()
+
+        renderer.render(template, renderResult, Map("users" -> Seq(Map("name" -> "Gigi"))))
+        renderResult.result.toString should be("<div>0-HelloGigi</div>")
```

Comment:
you should put at least two names in the context, so we would really see the indexes...

```html
<div>0-HelloGigi</div>
<div>1-HelloGicu</div>
```